### PR TITLE
IVsShellUtilitiesHelper cleanup

### PR DESCRIPTION
**Pull request by drewnoakes**
For a brief moment in time I was looking to extend this interface and its implementation to provide the VS installation directory. But it didn't quite do what I wanted.

Pulled these cleanup commits out to their own branch.

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Shell/Interop/IVsShellUtilitiesHelper.cs [hunk 1]
```diff
@@ -2,18 +2,24 @@
...
 using System;
 using System.Threading.Tasks;
+using Microsoft.VisualStudio.Composition;
+using Microsoft.VisualStudio.ProjectSystem;
 using Microsoft.VisualStudio.ProjectSystem.VS;
 using Microsoft.VisualStudio.ProjectSystem.VS.Interop;
 
 namespace Microsoft.VisualStudio.Shell.Interop
 {
+    [ProjectSystemContract(ProjectSystemContractScope.Global, ProjectSystemContractProvider.Private, Cardinality = ImportCardinality.ExactlyOne)]
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Shell/Interop/IVsShellUtilitiesHelper.cs]: This part that implements this imports IProjectThreadingService which lives in ProjectService scope. The unit test has caught that.

Comment (drewnoakes) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Shell/Interop/IVsShellUtilitiesHelper.cs]: Timely, given recent discussions. Will fix.

Comment (drewnoakes) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Shell/Interop/IVsShellUtilitiesHelper.cs]:
```suggestion
    [ProjectSystemContract(ProjectSystemContractScope.ProjectService, ProjectSystemContractProvider.Private, Cardinality = ImportCardinality.ExactlyOne)]
```

**Comment by davkean** [comment]
Unrelated, but these are basically extension methods, don't really need to be hidden behind interface, they should hide the VS interfaces, or just be converted to an extension method over the top of IVsXXX interface with the consumer switching before hand.

**Comment by drewnoakes** [comment]
The docs suggest this interface exists for unit testing purposes.

**Comment by davkean** [comment]
It probably was way more complicated at one point, it just wraps interfaces of which one has a default no op (switchtoUithread)

**Pull request merged**